themes/default/content/blog/build-images-50x-faster-docker-v4/index.md

```diff
@@ -0,0 +1,147 @@
+
+### Pulumi YAML and Pulumi Java support
+
+We announced [Pulumi YAML and Pulumi Java](https://www.pulumi.com/blog/pulumi-universal-iac/) last year and are excited to have extended support to the Pulumi Docker Provider. This release schematizes all Docker resources, which enables auto-support of all Pulumi languages. In addition, this means the Docker image resource can now participate in the CRUD (create, replace, update, delete) lifecycle.
+
+### Migrating to 4.0
+
+The 4.0 version of the Docker provider consists of a significant overhaul of the Image Resource.
+These changes are mainly a result of either the way Pulumi generates language SDKs from a schematized resource, or of migrating to using the docker-go client library instead of invoking the Docker CLI.
+A few additional changes are about streamlining the resource properties available.
+
+#### Inputs
+
+`LocalImageName`: Deprecated. Use `ImageName` in conjunction with `SkipPush:true`.
+
+`Build`: No longer a required input. No longer accepts a string for a custom build context. Use `Build.Context` instead.
+
+`ExtraOptions`: Deprecated. This was a way to pass arbitrary CLI flags to the Docker CLI.
+These are being implemented as top-level fields; currently available are:
+
+* `Build.Platform` - note that this can only take a single value currently, not a list of target platforms.
+* `Build.Target`
+* `Build.CacheFrom` - no longer accepts a boolean. Specify explicitly in images list:
+  * `Build.CacheFrom.Images` - replaces `Build.CacheFrom.Stages`, with updated functionality (a list of images to use as build cache)
+
+`Env`: Deprecated. This field’s use case was mainly for passing `DOCKER_BUILDKIT=1` to the CLI invocation of docker build. Use `Build.BuilderVersion` instead. You do not need to specify `Build.BuilderVersion` if you are running in Buildkit mode, which is the default.
+
+`ImageName`: Must be of fully qualified repository format, i.e. `registry.domain.abc/repositoryname/image`. The tag will be inferred as `:latest` unless otherwise specified. `Docker.io` will no longer be automatically inferred.
+
+#### Outputs
+
+`Id`: Deprecated. Use `ImageName` for a unique identifier.
+
+`Digest`: Deprecated as of an older version of pulumi-docker 3.6; this removes it entirely.
+
+#### Language specific package type updates
+
+{{< chooser language "python,csharp,go" >}}
+
+{{% choosable language csharp %}}
+
+```csharp
+Docker.ImageRegistry → Docker.Inputs.RegistryArgs
+Docker.DockerBuild → Docker.Inputs.DockerBuildArgs
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+docker.ImageRegistry → docker.Registry
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+DockerBuild → DockerBuildArgs
+ImageRegistry → RegistryArgs
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+The Docker Provider Image Resource has everything you need to build, run, and push Docker images for any container registry. Build your next image now!
```

Comment:
make this a new section such as "Getting started" and add a CTA that directs the user to getting started with it (e.g., docs link or registry link)

Reply:
@gthuang Updated. Please take a look at let me know if this change is better.